modules/basegl/src/processors/linerenderer.cpp

```diff
@@ -146,53 +130,36 @@ void LineRenderer::process() {
 }
 
 void LineRenderer::drawMeshes() {
-    bool autoDrawMode = (drawMode_.get() == LineDrawMode::Auto);
-    auto drawmode = util::getDrawMode(drawMode_.get(), useAdjacency_.get());
-
-    for (const auto& elem : inport_) {
-        if (elem->getNumberOfBuffers() == 0) continue;
-
-        auto& shader = lineShaders_.getShader(*elem);
-        shader.activate();
-        shader.setUniform("screenDim", vec2(outport_.getDimensions()));
-        utilgl::setUniforms(shader, camera_, lineWidth_, antialiasing_, miterLimit_, roundCaps_,
-                            stippling_);
-        MeshDrawerGL::DrawObject drawer(elem->getRepresentation<MeshGL>(),
-                                        elem->getDefaultMeshInfo());
-        utilgl::setShaderUniforms(shader, *elem, "geometry");
-        if (autoDrawMode) {
-            drawer.draw();
+    for (const auto& mesh : inport_) {
+        if (mesh->getNumberOfBuffers() == 0) continue;
+        MeshDrawerGL::DrawObject drawer(mesh->getRepresentation<MeshGL>(),
+                                        mesh->getDefaultMeshInfo());
+        if (mesh->getNumberOfIndicies() > 0) {
+            for (std::size_t i = 0; i < mesh->getNumberOfIndicies(); ++i) {
+                if (mesh->getIndexMeshInfo(i).dt != DrawType::Lines) continue;
```

Comment:
so one has to explicitly convert a tri mesh into lines before using this processor?

Reply:
Yes. I think we should optimise for the standard use case.